Issue using AssumeRoleProvider

I have a simple test that assume role provider works as I expect. The code is [here](https://github.com/data61/anonlink-entity-service/blob/59ba9d2f921cf15deb70ebfe95d11f1fd40e20f8/backend/entityservice/integrationtests/objectstoretests/test_objectstore.py). The main part of the code is:

```python
        root_mc_client = connect_to_object_store()
        upload_restricted_minio_client = connect_to_upload_object_store()
        if not root_mc_client.bucket_exists(bucket_name):
            root_mc_client.make_bucket(bucket_name)

        with pytest.raises(minio.S3Error):
            upload_restricted_minio_client.list_buckets()

        # Should be able to put an object though
        upload_restricted_minio_client.put_object(bucket_name, 'testobject', io.BytesIO(b'data'), length=4)

        credentials_provider = AssumeRoleProvider(
            f'http://{upload_endpoint}/',
            access_key=Config.UPLOAD_OBJECT_STORE_ACCESS_KEY,
            secret_key=Config.UPLOAD_OBJECT_STORE_SECRET_KEY,
            policy=restricted_upload_policy
        )

        credentials_provider.retrieve()
```

When I run it, a `ValueError` is raised:
```
>       credentials_provider.retrieve()

entityservice/integrationtests/objectstoretests/test_objectstore.py:61: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/site-packages/minio/credentials/providers.py:148: in retrieve
    res = _urlopen(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

http_client = <urllib3.poolmanager.PoolManager object at 0x7fb4affa4dc0>
method = 'POST', url = 'http://minio:9000/'
body = 'Action=AssumeRole&Version=2011-06-15&DurationSeconds=3600.0&Policy=%7B%0A++%22Version%22%3A+%222012-10-17%22%2C%0A++%...0%2F%2A%22%0A++++++%5D%2C%0A++++++%22Sid%22%3A+%22Upload-access-to-specific-bucket-only%22%0A++++%7D%0A++%5D%0A%7D+%0A'
headers = {'Authorization': 'AWS4-HMAC-SHA256 Credential=EXAMPLE_UPLOAD_ACCESS_KEY/20210209//sts/aws4_request, SignedHeaders=hos...0dd3212a', 'Content-Type': 'application/x-www-form-urlencoded', 'Host': 'minio:9000', 'X-Amz-Date': '20210209T101717Z'}

    def _urlopen(http_client, method, url, body=None, headers=None):
        """Wrapper of urlopen() handles HTTP status code."""
        res = http_client.urlopen(method, url, body=body, headers=headers)
        if res.status not in [200, 204, 206]:
>           raise ValueError(
                "{0} failed with HTTP status code {1}".format(url, res.status),
            )
E           ValueError: http://minio:9000/ failed with HTTP status code 400

/usr/lib/python3.8/site-packages/minio/credentials/providers.py:63: ValueError
```

This error message is not very helpful.

Running minio trace using the `mc` tool I can see the 400 bad request:
```
10:17:17.791 [200 OK] s3.GetBucketLocation minio:9000/uploads?location=  192.168.224.11    159µs       ↑ 93 B ↓ 349 B
10:17:17.792 [200 OK] s3.HeadBucket minio:9000/uploads 192.168.224.11    143µs       ↑ 93 B ↓ 189 B
10:17:17.793 [403 Forbidden] s3.ListBuckets minio:9000/ 192.168.224.11    375µs       ↑ 93 B ↓ 454 B
10:17:17.795 [200 OK] s3.PutObject minio:9000/uploads/testobject 192.168.224.11    2.861ms      ↑ 110 B ↓ 230 B
10:17:17.801 [400 Bad Request] sts.AssumeRole minio:9000/ 192.168.224.11    375µs       ↑ 533 B ↓ 490 B
```

In the minio server logs is the first clue:

```
API: AssumeRole()
Time: 10:25:16 UTC 02/09/2021
DeploymentID: 0aa63ef7-d5ee-4830-ab84-a6ae86e30804
RequestID: 16620D5E9EBBD133
RemoteHost: 192.168.240.11
Host: minio:9000
Error: invalid token expiry (*errors.errorString)
       5: cmd/sts-errors.go:57:cmd.writeSTSErrorResponse()
       4: cmd/sts-handlers.go:205:cmd.(*stsAPIHandlers).AssumeRole()
       3: cmd/http-tracer.go:204:cmd.Trace()
       2: cmd/handler-utils.go:356:cmd.httpTraceAll.func1()
       1: net/http/server.go:2042:http.HandlerFunc.ServeHTTP()
```

I'm using the docker image `minio/minio:RELEASE.2020-12-03T05-49-24Z` as the server. For comparison I have a [working version of the test code using minio-py v5](https://github.com/data61/anonlink-entity-service/blob/develop/backend/entityservice/integrationtests/objectstoretests/test_objectstore.py).

Perhaps relevant, the minio user was created with:

```bash
$ mc admin user add minio $UPLOAD_ONLY_ACCESS_KEY $UPLOAD_ONLY_SECRET_ACCESS_KEY
$ mc admin policy set minio writeonly user=$UPLOAD_ONLY_ACCESS_KEY
```

--- a/minio/credentials/providers.py
+++ b/minio/credentials/providers.py
@@ -98,11 +98,11 @@ class AssumeRoleProvider(Provider):
         query_params = {
             "Action": "AssumeRole",
             "Version": "2011-06-15",
-            "DurationSeconds": str(
+            "DurationSeconds": str(int(
                 duration_seconds
                 if duration_seconds > _DEFAULT_DURATION_SECONDS
                 else _DEFAULT_DURATION_SECONDS
-            ),
+            )),
         }
 
         if role_arn:

--- a/minio/credentials/providers.py
+++ b/minio/credentials/providers.py
@@ -37,9 +37,9 @@ from minio.xml import find, findtext
 
 from .credentials import Credentials
 
-_MIN_DURATION_SECONDS = timedelta(minutes=15).total_seconds()
-_MAX_DURATION_SECONDS = timedelta(days=7).total_seconds()
-_DEFAULT_DURATION_SECONDS = timedelta(hours=1).total_seconds()
+_MIN_DURATION_SECONDS = int(timedelta(minutes=15).total_seconds())
+_MAX_DURATION_SECONDS = int(timedelta(days=7).total_seconds())
+_DEFAULT_DURATION_SECONDS = int(timedelta(hours=1).total_seconds())
 
 
 def _parse_credentials(data, name):
@@ -98,11 +98,11 @@ class AssumeRoleProvider(Provider):
         query_params = {
             "Action": "AssumeRole",
             "Version": "2011-06-15",
-            "DurationSeconds": str(int(
+            "DurationSeconds": str(
                 duration_seconds
                 if duration_seconds > _DEFAULT_DURATION_SECONDS
                 else _DEFAULT_DURATION_SECONDS
-            )),
+            ),
         }
 
         if role_arn:

